### DISCUSSION TITLE
Documentation Questions

### DISCUSSION THREAD

**Discussion by random777DICE** [shakacode/react_on_rails#1666] [General]
Hi everyone,

I’m currently working on integrating React with my Rails application using react_on_rails. I have a couple of questions:

What are the best practices for setting up the initial configuration?
Has anyone encountered performance issues, and how did you resolve them?
Any tips for managing state effectively between React components and Rails?
I appreciate any insights or experiences you can share!

Thanks!

**Comment by justin808** [comment]
> What are the best practices for setting up the initial configuration?

See the spec/dummy folder and also https://github.com/shakacode/react-webpack-rails-tutorial/ for examples.

> Has anyone encountered performance issues, and how did you resolve them?

Too broad of a question. Profile and narrow down the problem. Backend? Frontend? etc.

> Any tips for managing state effectively between React components and Rails?

Again, too broad of a question.
